api/src/main/java/org/openmrs/module/idgen/processor/SequentialIdentifierGeneratorProcessor.java

```diff
@@ -58,15 +56,28 @@ public void setIdentifierSourceService(IdentifierSourceService identifierSourceS
     	
     	List<String> identifiers = new ArrayList<String>();
     	
-    	for (int i=0; i<batchSize;) {
-    		String val = seq.getIdentifierForSeed(sequenceValue);
-    		if (!reservedIdentifiers.contains(val)) {
-    			identifiers.add(val);
-    			i++;
-    		}
-	    	sequenceValue++;
-    	}
+    	if (seq instanceof LocationPrefixedSequentialIdentifierGenerator) {
+    		LocationPrefixedSequentialIdentifierGenerator locationPrefifixedGen = (LocationPrefixedSequentialIdentifierGenerator) seq;
+    		for (int i=0; i<batchSize;) {
+        		String val = locationPrefifixedGen.getIdentifierForSeed(sequenceValue);
+        		if (!reservedIdentifiers.contains(val)) {
+        			identifiers.add(val);
+        			i++;
+        		}
+    	    	sequenceValue++;
+        	}

```

Comment:
I believe that the above changes were not needed, and hence this file should probably not be touched. Am I missing something?